### DISCUSSION TITLE
Does message draining only work when the cluster is scaled?

### DISCUSSION THREAD

**Discussion by lavifb** [arkmq-org/arkmq-org-broker-operator#767] [Q&A] [answered]
Sometimes we may want to restart all the pods, like if we are upgrading to a new version of kubernetes and need to move to new nodes. Is there a way to maintain availability while restarting the pods one by one?

**Comment by gaohoward** [comment, marked as answer]
It should safe to restarting the pods without losing data.

**Reply by lavifb** [reply]
but are the messages drained to the other nodes? Or are they just saved in the pv for when the restarted node comes back?

**Reply by brusdev** [reply]
yes the messages saved in the pv  comes back when the node is restarted

**Comment by gaohoward** [comment]
make sure you enabled messageMigration (if persistenceEnabled is true then it by default enabled).
